### PR TITLE
ELY-2789 OIDCSecurityContext deserialization issue

### PR DESCRIPTION
**Pull request by PrarthonaPaul**
Issue: https://issues.redhat.com/browse/ELY-2789

### CODE REVIEW

File: http/oidc/src/test/java/org/wildfly/security/http/oidc/BearerTest.java [hunk 1]
```diff
@@ -146,6 +150,34 @@ public static void generalCleanup() throws Exception {
...
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteArrayOutputStream);
+        objectOutputStream.writeObject(oidcPrincipal);
+        objectOutputStream.close();
+
+        //deserialize
+        byte[] bytes = byteArrayOutputStream.toByteArray();
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(bytes);
+        ObjectInputStream objectInputStream = new ObjectInputStream(byteArrayInputStream);
+        OidcPrincipal deserialzedOidcPrincipal = (OidcPrincipal)objectInputStream.readObject();
```
Comment (fjuma) [inline http/oidc/src/test/java/org/wildfly/security/http/oidc/BearerTest.java]: deserializedOidcPrincipal

Comment (PrarthonaPaul) [reply http/oidc/src/test/java/org/wildfly/security/http/oidc/BearerTest.java]: fixed!

File: http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcSecurityContext.java [hunk 2]
```diff
@@ -76,8 +76,12 @@ public String getRealm() {
     private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
         in.defaultReadObject();
         try {
-            token = new AccessToken(new JwtConsumerBuilder().setSkipSignatureVerification().setSkipAllValidators().build().processToClaims(tokenString));
-            idToken = new IDToken(new JwtConsumerBuilder().setSkipSignatureVerification().setSkipAllValidators().build().processToClaims(idTokenString));
+            if (tokenString != null) {
+                token = new AccessToken(new JwtConsumerBuilder().setSkipSignatureVerification().setSkipAllValidators().build().processToClaims(tokenString));
```
Comment (fjuma) [inline http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcSecurityContext.java]: Perhaps if tokenString is null, we should set token to null. Same with the case below.

Comment (fjuma) [reply http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcSecurityContext.java]: e.g.,

`token = tokenString == null ? null : new AccessToken(new JwtConsumerBuilder().setSkipSignatureVerification().setSkipAllValidators().build().processToClaims(tokenString));`

Comment (PrarthonaPaul) [reply http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcSecurityContext.java]: fixed!

**Review by fjuma — APPROVED** [review]
Thanks @PrarthonaPaul!

**Review by ivassile — APPROVED** [review]

**Pull request merged**